source/extensions/dynamic_modules/sdk/rust/src/lib.rs

```diff
@@ -10,28 +10,45 @@ pub mod abi {
     include!(concat!(env!("OUT_DIR"), "/bindings.rs"));
 }
 
-/// Declare the init function for the dynamic module. This function is called when the dynamic module is loaded.
-/// The function must return true on success, and false on failure. When it returns false,
-/// the dynamic module will not be loaded.
+/// Declare the init functions for the dynamic module.
 ///
-/// This is useful to perform any process-wide initialization that the dynamic module needs.
+/// The first argument has [`ProgramInitFunction`] type, and it is called when the dynamic module is loaded.
+///
+/// The second argument has [`NewHttpFilterConfigFunction`] type, and it is called when the new HTTP filter configuration is created.
```

Comment:
nit: can you setup in some future PR rustfmt and make sure it runs in CI? This config https://github.com/bitdriftlabs/shared-core/blob/main/rustfmt.toml will make it mostly look like Envoy code style. For now you can run it manually.